### PR TITLE
Implemented JavaEntityBuilder

### LINKED ISSUE #39: feat: setup the builder for Java entities
- Create the EntityBuilder interface
- Implement the EntityBuilder interface in a JavaEntityBuilder class

Via the Builder design pattern

### PR DESCRIPTION
**Pull request by liamhyde02**
implemented JavaEntityBuilder according to builder pattern. Includes EntityBuilder interface, JavaEntityBuilder class implementing said interface, and JavaEntityType enumeration.

### CODE REVIEW

File: src/main/java/org/codevillage/JavaEntityBuilder.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    @Override
+    public void addComposition(String s) {
+        this.compositions.add(s);
+    }
+
+    @Override
+    public void addRealization(String s) {
+        this.realizations.add(s);
+    }
+
```
Comment (khoa-l) [inline src/main/java/org/codevillage/JavaEntityBuilder.java:66]: Should there be one for adding inheritances? Or is that covered under one of the other ones?

**Review by khoa-l — CHANGES_REQUESTED** [review]

File: src/main/java/org/codevillage/JavaEntityBuilder.java (hunk 1 above)
Comment (liamhyde02) [reply src/main/java/org/codevillage/JavaEntityBuilder.java:66]: I think inheritances are realizations? i can add a list of inheritances

Comment (khoa-l) [reply src/main/java/org/codevillage/JavaEntityBuilder.java:66]: wait isn't inheritance for abstract classes and classes and realization for interfaces? from list of requirements i saw: inheritance (abstract class / class), dependency (global variables), composition (constructor parameters), and realization (interfaces). I'm not sure the exact differences between those though so feel free to correct me

**Review by khoa-l — APPROVED** [review]
lgtm!

**Pull request merged**